src/foam/nanos/dig/DigSelectOperation.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.nanos.dig;
+
+import foam.core.*;
+import foam.dao.AbstractDAO;
+import foam.dao.ArraySink;
+import foam.dao.DAO;
+import foam.mlang.MLang;
+import foam.mlang.predicate.Predicate;
+import foam.nanos.http.*;
+import foam.util.SafetyUtil;
+
+import java.io.PrintWriter;
+import java.util.List;
+import javax.servlet.http.HttpServletResponse;
+
+public class DigSelectOperation extends DigOperation
```

Comment:
I would split these along the other axis. I would have DIGFormatDrivers (or something like that) and then have each driver support all of the operations required for that format: find, selelct, remove, etc. Then, if I wanted to add a new format it would be self-contained rather than having to update each operation class. If the driver were modelled, then it could also have format specific fields (ie. pretty-print, indent, quote-keys, etc.).